File: Backend/Domain/Models/ClientModel.cs
using Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace Domain.Models
{

    public class ClientModel
    {
        public Guid? Id { get; set; } = default!;
        [Required(ErrorMessage = "The PersonalNumber field is required.")]
        public string PersonalNumberId { get; set; } = default!;
        [Required(ErrorMessage = "The FirstName field is required.")]
        public string FirstName { get; set; } = default!;

        [Required(ErrorMessage = "The MiddleName field is required.")]
        public string MiddleName { get; set; } = default!;
        [Required(ErrorMessage = "The LastName field is required.")]
        public string LastName { get; set; } = default!;
        [Required(ErrorMessage = "The PhoneNumber field is required.")]
        public string PhoneNumber { get; set; } = default!;
        [Required(ErrorMessage = "The EmailAddress field is required.")]
        public string EmailAddress { get; set; } = default!;
        [Required(ErrorMessage = "The State field is required.")]
        public StateOfClient State { get; set; } = default!;
        [Required(ErrorMessage = "The City field is required.")]
        public CityOfClient City { get; set; } = default!;
        [Required(ErrorMessage = "The ZipCode field is required.")]
        public string ZipCode { get; set; } = default!;
    }
}

File: Frontend/src/interfaces/client-model.ts
import { StateOfClient } from "./StateOfClient";
import { CityOfClient } from "./CityOfClient";

export interface ClientModel{
    id: string | null;
    personalNumberId:string | null;
    firstName:string |null;
    middleName:string |null;
    lastName:string |null;
    phoneNumber:string|null;
    emailAddress:string | null;
    state:StateOfClient;
    city:CityOfClient;
    zipCode:string| null;
    
}